multiview/embed/gcca.py

```diff
@@ -0,0 +1,249 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .base import BaseEmbed
+from ..utils.utils import check_Xs
+
+import numpy as np
+from scipy import linalg, stats
+from scipy.sparse.linalg import svds
+from sklearn.preprocessing import normalize
+
+
+class GCCA(BaseEmbed):
+    """
+    An implementation of Generalized Canonical Correalation Analysis. Computes
+    individual projections into a common subspace such that the correlations
+    between pairwise projections are minimized (ie. maximize pairwise
+    correlation). Reduces to CCA in the case of two samples.
+
+    See https://www.sciencedirect.com/science/article/pii/S1053811912001644?
+    via%3Dihub
+    for relevant details.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._projection_mats = None
+
+    def _center(self, X):
+        """
+        Subtracts the row means and divides by the row standard deviations.
+        Then subtracts column means.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_observations, n_features)
+            The data to preprocess
+
+        Returns
+        -------
+        centered_X : preprocessed data matrix
+        """
+
+        # Mean along rows using sample mean and sample std
+        centered_X = stats.zscore(X, axis=1, ddof=1)
+        # Mean along columns
+        mu = np.mean(centered_X, axis=0)
+        centered_X -= mu
+        return centered_X
+
+    def fit(
+        self,
+        Xs,
+        fraction_var=0.9,
+        sv_tolerance=None,
+        n_components=None,
```

Comment:
if you guys are going for sklearn style then all of this needs to be attributes of the class, and not passed into `fit` @rflperry @rguo123